atomic_sequence.go

```diff
@@ -0,0 +1,281 @@
+package sequence
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync/atomic"
+)
+
+// NewAtomic creates a NewSequence that implements the Incrementer interface.
+// It is safe for concurrent use.
+func NewAtomic(params ...uint64) (*AtomicSequence, error) {
```

Comment:
Would it be useful to have `New()` take a flag that returns `*AtomicSequence` if true or `*Sequence` by default? E.g.: 

```go
func New(atomic bool, params ...uint64) (Incrementer, error) {
	...
}
```

Reply:
I think it would! I am not sure though because when you see it from outside:
`go
New(false, 1, 100, 150)
`
It doesn't say much IMO. Perhaps with a constant like:
`
type SequenceType uint8
const (
Default SequenceType = iota
Atomic
)

// elsewhere
New(sequence.Default, params...)
New(sequence.Atomic, params...)
`

would make it a little better on the eyes but I am not sure if the "complexity" added there is needed though

Reply:
Yeh, that's a tricky one ... let's just bump it for now!

Reply:
-->